dialect/pgdialect/range.go

```diff
@@ -1,104 +1,202 @@
+}
+
+func (r *Range[T]) IsZero() bool {
+	return r == nil || r.LowerBound == 0
+}
+
+func (r Range[T]) IsEmpty() bool {
+	return r.LowerBound == RangeBoundEmpty
 }
 
 var _ sql.Scanner = (*Range[any])(nil)
 
-func (r *Range[T]) Scan(anySrc any) (err error) {
-	src, ok := anySrc.([]byte)
-	if !ok {
-		return fmt.Errorf("pgdialect: Range can't scan %T", anySrc)
+func (r *Range[T]) Scan(raw any) (err error) {
+	var src []byte
+	switch v := raw.(type) {
+	case []byte:
+		src = v
+	case string:
+		src = []byte(v)
+	case nil:
+		return nil
+	default:
+		return fmt.Errorf("pgdialect: Range can't scan %T", raw)
 	}
 
+	src = bytes.TrimSpace(src)
 	if len(src) == 0 {
-		return io.ErrUnexpectedEOF
+		return nil
 	}
-	r.LowerBound = RangeBound(src[0])
-	src = src[1:]
 
-	src, err = scanElem(&r.Lower, src)
-	if err != nil {
-		return err
+	if string(src) == "empty" {
+		r.LowerBound, r.UpperBound = RangeBoundEmpty, RangeBoundEmpty
+		return nil
 	}
 
-	if len(src) == 0 {
-		return io.ErrUnexpectedEOF
+	switch src[0] {
+	case byte(RangeBoundInclusiveLeft), byte(RangeBoundExclusiveLeft):
+		r.LowerBound = RangeBound(src[0])
+	default:
+		return fmt.Errorf("unexpected lower bound: %s", string(src[:1]))
 	}
-	if ch := src[0]; ch != ',' {
-		return fmt.Errorf("got %q, wanted %q", ch, ',')
+	switch src[len(src)-1] {
+	case byte(RangeBoundInclusiveRight), byte(RangeBoundExclusiveRight):
+		r.UpperBound = RangeBound(src[len(src)-1])
+	default:
+		return fmt.Errorf("unexpected upper bound: %s", string(src[len(src)-1:]))
 	}
-	src = src[1:]
 
-	src, err = scanElem(&r.Upper, src)
-	if err != nil {
-		return err
+	src = src[1 : len(src)-1]
+
+	ind := bytes.IndexByte(src, ',')
+	if ind == -1 {
+		return fmt.Errorf("invalid range: wanted comma, got %s", string(src))
 	}
+	left, right := src[:ind], src[ind+1:]
 
-	if len(src) == 0 {
-		return io.ErrUnexpectedEOF
+	if len(left) > 0 {
+		_, err := scanElem(&r.Lower, left)
+		if err != nil {
+			return err
+		}
+	} else {
+		r.LowerBound = RangeBoundUnbound
 	}
-	r.UpperBound = RangeBound(src[0])
-	src = src[1:]
 
-	if len(src) > 0 {
-		return fmt.Errorf("unread data: %q", src)
+	if len(right) > 0 {
+		_, err = scanElem(&r.Upper, right)
+		if err != nil {
+			return err
+		}
+	} else {
+		r.UpperBound = RangeBoundUnbound
 	}
+
 	return nil
 }
 
 var _ schema.QueryAppender = (*Range[any])(nil)
 
-func (r *Range[T]) AppendQuery(fmt schema.Formatter, buf []byte) ([]byte, error) {
-	buf = append(buf, byte(r.LowerBound))
-	buf = appendElem(buf, r.Lower)
+func (r Range[T]) AppendQuery(_ schema.Formatter, buf []byte) ([]byte, error) {
+	buf = append(buf, '\'')
+	buf = appendRange(buf, r)
+	buf = append(buf, '\'')
+	return buf, nil
+}
+
+func appendRange[T any](buf []byte, r Range[T]) []byte {
+	if r.IsEmpty() {
+		buf = append(buf, []byte("empty")...)
+		return buf
+	}
+
+	if r.LowerBound == RangeBoundUnbound {
+		buf = append(buf, byte(RangeBoundExclusiveLeft))
```

Comment:
I don't understand why we use RangeBoundExclusiveLeft if it is unbound. May be add a comment?